modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaBaseModelListenerCheck.java

```diff
@@ -32,41 +35,79 @@ protected String format(
 
 		JavaClass javaClass = JavaClassParser.parseJavaClass(fileName, content);
 
-		List<String> implementedClassNames =
-			javaClass.getImplementedClassNames();
+		List<String> extendedClassNames = javaClass.getExtendedClassNames();
 
-		if (implementedClassNames.contains("BaseModelListener")) {
+		if (!extendedClassNames.contains("BaseModelListener")) {
 			return content;
 		}
 
+		_extendedClassType = _getExtendedClassType(content);
+
 		for (JavaTerm childJavaTerm : javaClass.getChildJavaTerms()) {
 			if (!childJavaTerm.isJavaMethod()) {
 				continue;
 			}
 
 			JavaMethod javaMethod = (JavaMethod)childJavaTerm;
 
-			String newJavaMethodContent = _formatMethodDefinition(javaMethod);
+			String newJavaContent = _formatMethod(javaMethod);
 
 			content = StringUtil.replace(
-				content, javaMethod.getContent(), newJavaMethodContent);
+				content, javaMethod.getContent(), newJavaContent);
 		}
 
 		return content;
 	}
 
-	private String _formatMethodDefinition(JavaMethod javaMethod) {
-		String javaMethodContent = javaMethod.getContent();
-
+	private String _formatMethod(JavaMethod javaMethod) {
 		String javaMethodName = javaMethod.getName();
 
-		if (!javaMethodName.equals("onAfterUpdate") &&
-			!javaMethodName.equals("onBeforeUpdate")) {
+		if (javaMethodName.equals("onAfterUpdate") ||
+			javaMethodName.equals("onBeforeUpdate")) {
+
+			return _formatMethodDefinitionWithParameterUpgrade(javaMethod);
+		}
+
+		return _formatMethodDefinitionWithoutParameterUpgrade(javaMethod);
+	}
 
+	private String _formatMethodDefinitionWithoutParameterUpgrade(
+		JavaMethod javaMethod) {
+
+		String javaMethodContent = javaMethod.getContent();
+
+		Matcher matcher = _superMethodPattern.matcher(javaMethodContent);
+
+		if (!matcher.find()) {
 			return javaMethodContent;
 		}
 
-		javaMethodContent = _formatSuper(javaMethodContent);
+		String methodCall = JavaSourceUtil.getMethodCall(
+			javaMethodContent, matcher.start());
+
+		methodCall = methodCall.trim();
+
+		String parameterName = JavaSourceUtil.getParameters(methodCall);
+
+		String cloneMethod = StringBundler.concat(
+			_extendedClassType, " original",
+			StringUtils.capitalize(parameterName), " = (", _extendedClassType,
+			")model.clone();");
+
+		String spacing = matcher.group(1);
```

Comment:
This might technically violate a soft SF rule of not using matcher.group(1) directly, but naming it like this makes it way more readable than putting in matcher.group(1) in a StringUtil.replace and require the reading of regex, even if it is simple

Reply:
Use String.format instead to just make it readable

Reply:
Let's drop `spacing` for simplicity. SF can catch bad formatting later.